core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java

```diff
@@ -68,12 +69,23 @@
 
   private Configuration conf;
 
+  private FileIO fileIO;
+
   public HadoopTables() {
     this(new Configuration());
   }
 
+  public HadoopTables(FileIO fileIO) {
```

Comment:
Could you please explain why do we need the other `FileIO` rather than initializing the `HadoopFileIO` by default ?

Reply:
Thanks for your review, My FileSystem which has some external function such as authentication is not a standard implementation. In other cases, Many Engines has owner FileSystem Manager so they don't need HadoopTables to initialize FileSystem.

Reply:
If the `FileSystem` is not a standard HDFS implementation, then I will think the `table` is not a stand hadoop table. So why not just define the customized catalog & table implementation ?

Reply:
In the one hand, The only difference is the FileIO. The constructor of HadoopTableOperations is protected and constructor args has FileIO, So the  design philosophy of HadoopTableOperations is support Dynamic FileIO. 

In other hand, like apache/iceberg#1618